receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/factory.go

```diff
@@ -54,6 +55,11 @@ func (f *Factory) CreateMetricsScraper(
 	config internal.Config,
 ) (scraperhelper.Scraper, error) {
 	cfg := config.(*Config)
+
+	if _, err := os.Stat("/.dockerenv"); cfg.RootPath == "" && err != nil {
```

Comment:
Is this a reliable way to detect the container environment? What about other container engines like contained?

Reply:
Reading https://github.com/moby/moby/blob/598c2957075819bcb19e8523258ad41cc50d0949/daemon/initlayer/setup_unix.go#L29, it looks like this file is mounted regardless of the OCI manager used.

Reply:
I also realised that `podman` would start the container differently, so I have added a check for their default mounts.